test/EndToEnd/e2e_tests.mk

```diff
@@ -0,0 +1,48 @@
+# This file contains the make targets for building and pushing e2e test app images.
+
+E2E_TEST_APPS_DIR := ./test/EndToEnd/Apps
+
+# E2E_TEST_APPS contains a list of all e2e test apps.
+# This list is used to build and push all e2e test app images.
+E2E_TEST_APPS=$(shell ls $(E2E_TEST_APPS_DIR))
+
+# check the required environment variables
+check-e2e-env:
+ifeq ($(DAPR_E2E_TEST_APP_REGISTRY),)
+	$(error DAPR_E2E_TEST_APP_REGISTRY environment variable must be set)
+endif
+ifeq ($(DAPR_E2E_TEST_APP_TAG),)
+	$(error DAPR_E2E_TEST_APP_TAG environment variable must be set)
+endif
+
+define genTestAppImageBuild
+.PHONY: build-e2e-app-$(1)
+build-e2e-app-$(1): check-e2e-env
+	@echo "Building e2e test app $(1) image"
+	docker buildx build --platform=linux/amd64 --output type=docker -t $(DAPR_E2E_TEST_APP_REGISTRY)/$(1):$(DAPR_E2E_TEST_APP_TAG) -f $(E2E_TEST_APPS_DIR)/$(1)/Dockerfile .
```

Comment:
Where are we saving the docker image? We need to login to the registry for pushing the image, right?

Reply:
Right, check the README for e2e, the registry and tag needs to be provided by the user, the image gets pushed to it first and then pulled during tests.